tests/foreman/maintain/test_health.py

```diff
@@ -0,0 +1,827 @@
+        elif name in ['foreman_repo', 'puppet_repo']:
+            assert 'enabled=0' in result.stdout
+
+    @request.addfinalizer
+    def _finalize():
+        for name, url in upstream_url.items():
+            sat_maintain.execute(f'rm -fr /etc/yum.repos.d/{name}.repo')
+        sat_maintain.execute('yum clean all')
+
+
+@pytest.mark.include_capsule
+def test_positive_health_check_available_space(sat_maintain):
+    """Verify available-space check
+
+    :id: 7d8798ca-3334-4dda-a9b0-dc3d7c0903e9
+
+    :parametrized: yes
+
+    :steps:
+        1. Run satellite-maintain health check --label available-space
+
+    :expectedresults: available-space health check should pass.
+    """
+    result = sat_maintain.cli.Health.check(options={'label': 'available-space'})
+    assert 'FAIL' not in result.stdout
+    assert result.status == 0
+
+
+def test_positive_health_check_available_space_candlepin(sat_maintain):
+    """Verify available-space-cp check
+
+    :id: 382a2bf3-a3da-4e46-b370-a443450f93b7
+
+    :steps:
+        1. Run satellite-maintain health check --label available-space-cp
+
+    :expectedresults: available-space-cp health check should pass.
+    """
+    result = sat_maintain.cli.Health.check(options={'label': 'available-space-cp'})
+    assert 'FAIL' not in result.stdout
+    assert result.status == 0
+
+
+def test_positive_hammer_defaults_set(sat_maintain, request):
+    """Verify that health check is performed when
+     hammer on system have defaults set.
+
+    :id: 27a8b49b-8cb8-4004-ba41-36ed084c4740
+
+    :steps:
+        1. Setup hammer on system with defaults set
+        2. Run satellite-maintain health check
+
+    :expectedresults: Health check should pass.
+
+    :BZ: 1632768
+
+    :customerscenario: true
+    """
+    sat_maintain.cli.Defaults.add({'param-name': 'organization_id', 'param-value': 1})
+    result = sat_maintain.cli.Health.check(options={'assumeyes': True})
+    assert result.status == 0
+    assert 'FAIL' not in result.stdout
+
+    @request.addfinalizer
+    def _finalize():
+        sat_maintain.cli.Defaults.delete({'param-name': 'organization_id'})
+
+
+@pytest.mark.include_capsule
+def test_positive_health_check_hotfix_installed(sat_maintain, request):
+    """Verify check-hotfix-installed check.
+
+    :id: d9023293-4173-4223-bbf5-328b41cf87cd
+
+    :parametrized: yes
+
+    :setup:
+        1. Modify gems files of a Satellite
+        2. Install hotfix-package
+
+    :steps:
+        1. Run satellite-maintain health check --label check-hotfix-installed
+
+    :expectedresults: check-hotfix-installed check should detect modified file
+        and installed hotfix.
+    """
+    # Verify check-hotfix-installed without hotfix package.
+    result = sat_maintain.cli.Health.check(options={'label': 'check-hotfix-installed'})
+    assert result.status == 0
+    assert 'WARNING' not in result.stdout
+
+    # Verify check-hotfix-installed with hotfix package.
+    gems_path = '/usr/share/gems/gems/'
```

Comment:
This path will change for el7 and el8
```suggestion
    gems_path = f"""{f"{'/opt/theforeman/tfm/root' if sat_maintain.os_version.major < 8 else ''}" + '/usr/share/gems/gems/'}"""
```

Reply:
Updated.